src/core/lib/iomgr/tcp_posix.c

```diff
@@ -107,7 +112,16 @@ static void tcp_handle_write(grpc_exec_ctx *exec_ctx, void *arg /* grpc_tcp */,
 
 static void tcp_shutdown(grpc_exec_ctx *exec_ctx, grpc_endpoint *ep) {
   grpc_tcp *tcp = (grpc_tcp *)ep;
-  grpc_fd_shutdown(exec_ctx, tcp->em_fd);
+  gpr_mu_lock(&tcp->mu);
```

Comment:
What happens if tcp_write or tcp_read are called concurrently or after tcp_shutdown?
Ideally we'd want them to fail immediately, currently I suspect we'd see some kind of crash.